using_restricted_array.rb

```diff
@@ -4,44 +4,132 @@
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
+# Implement the methods in using_restricted_array.rb. Methods to implement are:
+#
+# length - which calculates the length of the integer array and returns it. Note: The restricted_array is terminated by nil i.e. array[length] = nil
+
+# print_array - which prints each integer value separated by space on the same line
+
+# reverse - which reverses the values in the array in place
+
+# search - which looks for a given integer value in the array. Returns true if found, false otherwise. Note: do not assume that the array is sorted.
+# binary_search - which looks for a given integer value in the array. Returns true if found, false otherwise. Note: Assume that the array is sorted in ascending order.
+# find_largest - Finds and returns the largest value element in the integer array.
+# find_smallest - Finds and returns the smallest value element in the integer array.
+
+
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  array_values = []
```

Comment:
Using array_values makes your space complexity O(n). If instead of adding to array_values, you just print array[i], then you'll have a space complexity of O(1).